### PR TITLE
Convert length and some angle units in and out of SI

### LINKED ISSUE #3: Localisation_Toolkit: Add initial converts
Convert doubles in and out of SI units. There should only be SI converts as discussed with @IsakNaslundBh , so if you want to convert mm to inches a daisy chain of methods will be needed to convert mm to SI and SI to inches.

### PR DESCRIPTION
**Pull request by FraserGreenroyd**
### Issues addressed by this PR
Fixes #3 

 ### Test files

 ### Changelog
 - Convert mm to SI and SI to mm
 - Convert cm to SI and SI to cm
 - Convert km to SI and SI to km
 - Convert micrometer to SI and SI to micrometer
 - Convert nanometer to SI and SI to nanometer
 - Convert Angstrom to SI and SI to Angstrom
 - Convert mile to SI and SI to mile
 - Convert yard to SI and SI to yard
 - Convert foot to SI and SI to foot
 - Convert inch to SI and SI to inch
 - Convert Arcminute to SI and SI to Arcminute
 - Convert Arcsecond to SI and SI to Arcsecond

 ### Additional comments
Would aim to put this into an alpha installer asap after merging if works to design, FYI @al-fisher :smile:

### CODE REVIEW

**Comment by IsakNaslundBh** [comment]
Just a first comment, before checking through the methods, but think it might be worth to put all the files already added into a "Length" folder? Afraid this can become completely unreadable if we keep every single unit type for every single dimension as a flat list

**Comment by FraserGreenroyd** [comment]
@IsakNaslundBh done, I did wonder about that myself but wasn't sure, glad you said the same as me :smile:

**Comment by FraserGreenroyd** [comment]
I have made the changes you requested @IsakNaslundBh and requested review from both of you again.

Personally, given the discussion we had around wrapping UnitsNet, I'm thinking these might only be very temporary files until the wrapping is completed anyhow, and these won't be going into an installer anytime soon so can say use at own risk on people using from source in the interim :smile:

File: Localisation_Engine/Convert/Length/Mile.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+namespace BH.Engine.Localisation.Length
+{
+    public static partial class Convert
+    {
+        [Description("Convert mile into SI units (meter)")]
+        [Input("mile", "Mile measurement")]
+        [Output("si", "Mile represented as SI units (meter)")]
+        public static double FromMile(this double mile)
+        {
+            return mile * 1609.347;
```
Comment (JosefTaylor) [inline Localisation_Engine/Convert/Length/Mile.cs]: 0.3048 * 5280 = 1609.344

Comment (FraserGreenroyd) [reply Localisation_Engine/Convert/Length/Mile.cs]: That's what I had it before I listened to @IsakNaslundBh 🤦‍♂ 
I'll change it back, gimme 1 minute!

**Comment by FraserGreenroyd** [comment]
@JosefTaylor have reverted the mile change if you could take another look when you have a moment :smile:

**Pull request merged**